axelrod/tests/strategies/test_meta.py

```diff
@@ -585,3 +585,77 @@ def test_strategy(self):
         actions = [(C, C), (C, D), (D, C), (D, D), (D, C)]
         self.versus_test(opponent=axelrod.Alternator(),
                          expected_actions=actions)
+
+"""Tests for the Memory Decay strategy"""
+
+class TestMemoryDecay(TestPlayer):
+
+    name = "Memory Decay: 0.1, 0.03, -2, 1, <class 'axelrod.strategies.titfortat.TitForTat'>, 15"
```

Comment:
This is a bit of a mouthful but I think it's fine.

Reply:
Could we look at what happens if we modify the `axl.Player` `__repr__` method to be:

```
  196     def __repr__(self):                                                                                                                         
  197         """The string method for the strategy.                                                                                                  
  198         Appends the `__init__` parameters to the strategy's name."""                                                                            
  199         name = self.name                                                                                                                        
  200         prefix = ': '                                                                                                                           
  201         gen = (value for value in self.init_kwargs.values() if value is not None)                                                               
  202         for value in gen:                                                                                                                       
+ 203             if issubclass(value, Player):                                                                                                       
+ 204                 value = value.name                                                                                                              
  205             name = ''.join([name, prefix, str(value)])                                                                                          
  206             prefix = ', '                                                                                                                       
  207         return name   
```

(Lines 203, 204 of https://github.com/Axelrod-Python/Axelrod/blob/master/axelrod/player.py, will need a `from axelrod.player import Player` at the top.)

This way it ensures the default player repr displays the strategy name and not the class...

Reply:
Sorry that won't work. This does:

```
  195     def __repr__(self):                                                                                                                         
  196         """The string method for the strategy.                                                                                                  
  197         Appends the `__init__` parameters to the strategy's name."""                                                                            
  198         name = self.name                                                                                                                        
  199         prefix = ': '                                                                                                                           
  200         gen = (value for value in self.init_kwargs.values() if value is not None)                                                               
  201         for value in gen:                                                                                                                       
+ 202             try:                                                                                                                                
+ 203                 if issubclass(value, Player):                                                                                                   
+ 204                     value = value.name                                                                                                          
+ 205             except TypeError:                                                                                                                   
+ 206                 pass                                                                                                                            
  207             name = ''.join([name, prefix, str(value)])                                                                                          
  208             prefix = ', '                                                                                                                       
  209         return name  
```